charts/linkerd2/templates/destination.yaml

```diff
@@ -131,6 +131,7 @@ spec:
         {{ include "partials.annotations.created-by" . }}
         {{- include "partials.proxy.annotations" . | nindent 8}}
         {{- with .Values.podAnnotations }}{{ toYaml . | trim | nindent 8 }}{{- end }}
+        config.linkerd.io/opaque-ports: {{.Values.proxy.opaquePorts}}
```

Comment:
It shouldn't belong to `partials.annotations` because now that we add the annotation to services, we don't also want to include the other annotations in there.

I think we could maybe define a `partials.annotations.config` (or something like that) that appends additional proxy configurations, but not until we have at least another annotation that we add similar to `opaque-ports`. It would ultimately be the same number of lines in the diff to explicitly at the `opaque-ports` annotation or this new `partials.annotations.config`.